services/src/main/scala/cromwell/services/cost/GcpCostCatalogService.scala

```diff
@@ -212,23 +235,47 @@ class GcpCostCatalogService(serviceConfig: Config, globalConfig: Config, service
     }
 
   // TODO consider caching this, answers won't change until we reload the SKUs
-  def calculateVmCostPerHour(instantiatedVmInfo: InstantiatedVmInfo): ErrorOr[BigDecimal] =
-    for {
+  def calculateVmCostPerHour(instantiatedVmInfo: InstantiatedVmInfo): ErrorOr[BigDecimal] = {
+    val cpuPricingInfoErrorOr = for {
       cpuSku <- lookUpSku(instantiatedVmInfo, Cpu)
       coreCount <- MachineType.extractCoreCountFromMachineTypeString(instantiatedVmInfo.machineType)
       cpuPricePerHour <- GcpCostCatalogService.calculateCpuPricePerHour(cpuSku, coreCount)
+    } yield (cpuSku, coreCount, cpuPricePerHour)
+
+    val ramPricingInfoErrorOr = for {
       ramSku <- lookUpSku(instantiatedVmInfo, Ram)
       ramMbCount <- MachineType.extractRamMbFromMachineTypeString(instantiatedVmInfo.machineType)
       ramGbCount = ramMbCount / 1024d // need sub-integer resolution
       ramPricePerHour <- GcpCostCatalogService.calculateRamPricePerHour(ramSku, ramGbCount)
-      totalCost = cpuPricePerHour + ramPricePerHour
+    } yield (ramSku, ramGbCount, ramPricePerHour)
+
+    val gpuPricingInfoErrorOr = instantiatedVmInfo.gpuInfo match {
+      case None => (None, 0, BigDecimal(0)).validNel
+      case Some(gpuInfo) =>
+        for {
+          gpuSku <- lookUpSku(instantiatedVmInfo, Gpu)
+          gpuCount = gpuInfo.count
+          gpuPricePerHour <- GcpCostCatalogService.calculateGpuPricePerHour(gpuSku, gpuCount)
+        } yield (Some(gpuSku), gpuCount, gpuPricePerHour)
+    }
+
+    for {
+      cpuPricingInfo <- cpuPricingInfoErrorOr
+      (cpuSku, coreCount, cpuPricePerHour) = cpuPricingInfo
```

Comment:
The downside of breaking up this method into smaller `for-yield`s is that, since the log message at the end incorporates information from nearly every line of every `for-yield`, we need to box up the results of all the `for-yield`s into `ErrorOr`s of tuples and then unravel all the data here. It would make me happy if, when I think to check back on this PR a month from now, I see that someone with more Scala knowledge than me was able to figure out a way to make this less messy!